src/components/IconWithTooltip/index.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { OverlayTrigger, Tooltip } from '@edx/paragon';
+import useWindowSize from '../../hooks/useWindowSize';
+
+
+const IconWithTooltip = ({
+  icon, altText, tooltipText, placementSm = 'right', placementLg = 'top', trigger = ['hover', 'focus'], breakpoint = 768, iconClassNames = 'ml-1',
+}) => {
+  const windowSize = useWindowSize();
+  return (
+    <OverlayTrigger
+      trigger={trigger}
+      placement={windowSize.width >= breakpoint ? placementSm : placementLg}
+      overlay={
+        <Tooltip id="tooltip-right">
+          {tooltipText}
+        </Tooltip>
+    }
+    >
+      <FontAwesomeIcon className={iconClassNames} icon={icon} alt={altText} />
+    </OverlayTrigger>
+  );
+};
+
+IconWithTooltip.propTypes = {
+  icon: PropTypes.shape({}).isRequired,
+  altText: PropTypes.string.isRequired,
+  tooltipText: PropTypes.string.isRequired,
+  // These props have defaults above
+  /* eslint-disable react/require-default-props */
+  placementSm: PropTypes.oneOf(['top', 'right', 'left', 'bottom']),
+  placementLg: PropTypes.oneOf(['top', 'right', 'left', 'bottom']),
+  trigger: PropTypes.oneOfType(PropTypes.string, PropTypes.array),
+  breakpoint: PropTypes.number,
+  iconClassNames: PropTypes.string,
+  /* eslint-enable react/require-default-props */
```

Comment:
not sure what this is and why we need it?

Reply:
So.... the eslint settings we have aren't quite right for function components. These props *have* defaults, in the code above, but I'm not using `defaultPropTypes`. Mostly because I think it's easier to read with the default where it currently is. We could turn the eslint setting off for function components; but that's not quite what I want to do either; I *want* these to have defaults; I just want them in a different location.

Reply:
ah ok just never used this rule so was curious what it was about. looks like you got it! just maybe want to add a line on why this was added would be cool